### PR TITLE
Bump OTEL collector to 0.42.0

### LINKED ISSUE #658: Release 0.42.0
OTEL collector 0.42.0 has been released, we should release the operator and add an upgrade procedure if needed.

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <p.loffay@gmail.com>

Resolves #658

### CODE REVIEW

**Comment by pavolloffay** [comment]
@jpkrohling you might be the best one to review this one.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 Changes by Version
 ==================
 
+0.42.0
```
Comment (VineethReddy02) [inline CHANGELOG.md:4]: In the release PR, we only add changes that matter to end-users into `CHANGELOG.md`, not all the changes or improvements that don't matter to the end-user. 
Also, you should include `* Bumped OpenTelemetry Collector to v0.42.0  in CHANGELOG. `

Comment (pavolloffay) [reply CHANGELOG.md:4]: Which change does not matter? I find image changes useful to know.

Comment (VineethReddy02) [reply CHANGELOG.md:4]: I mean this isn't end-user related IMO `Parse flags before using them in config ([#662](https://github.com/open-telemetry/opentelemetry-operator/pull/662), [@rubenvp8510](https://github.com/rubenvp8510))`

Comment (pavolloffay) [reply CHANGELOG.md:4]: it's a fix that fixes paring of the flags which was broken

**Review by VineethReddy02 — APPROVED** [review]
LGTM!

As it's the release PR and breaking changes from collector and contrib needs to be verified. I have taken a quick look over release notes didn't find any changes that need an upgrade routine at the operator end. 

cc @jpkrohling for confirmation/otel version bump approval.

**Comment by jpkrohling** [comment]
I trust both you and @pavolloffay have looked into the changelog for the operands and worked on the upgrade routines. Let's ship it :-)

**Pull request merged**